File: examples/objc/micro/UserService.m
#import "UserService.h"

@implementation UserService
- (NSNumber *)getUser:(NSString *)token {
    AuthService *auth = [[AuthService alloc] init];
    return [auth verifyToken:token];
}

- (NSNumber *)updateUser:(NSString *)token action:(NSString *)action {
    AuthService *auth = [[AuthService alloc] init];
    NSNumber *userId = [auth verifyToken:token];
    if (userId != nil) {
        [auth runAdminCommand:userId action:action];
    }
    return userId;
}
@end

File: examples/lua/micro/user_service.lua
local auth = require("auth_service")

local M = {}

function M.getUser(token)
    return auth.verifyToken(token)
end

function M.updateUser(token, action)
    local user_id = auth.verifyToken(token)
    if user_id then
        auth.runAdminCommand(user_id, action)
    end
    return user_id
end

return M
